pkgs/applications/editors/quilter/default.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv, fetchFromGitHub, fetchpatch, vala, pkgconfig, meson, ninja, granite
+, gtk3, desktop-file-utils, gnome3, gtksourceview, webkitgtk, gtkspell3, discount
+, gobjectIntrospection, wrapGAppsHook }:
+
+stdenv.mkDerivation rec {
+  pname = "quilter";
+  version = "1.6.3";
+
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "lainsce";
+    repo = pname;
+    rev = version;
+    sha256 = "1wa0i6dgg6fgb7q9z33v9qmn1a1dn3ik58v1f3a49dvd5xyf8q6q";
+  };
+
+  nativeBuildInputs = [
+    desktop-file-utils
+    gobjectIntrospection
+    meson
+    ninja
+    pkgconfig
+    vala
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    discount
+    granite
+    gtk3
+    gtksourceview
+    gtkspell3
+    webkitgtk
+    gnome3.libgee
+  ];
+
+  LIBRARY_PATH = stdenv.lib.makeLibraryPath [ discount ];
+
+  patches = [
+    # Fixes build with vala 0.42
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/decathorpe/elementary-appcenter-rpms/master/sources/quilter/00-fix-vala-042-errors.patch";
+      sha256 = "1wnq3div72bacxcvviqr2c6c8nhmlqrqz33j645sfmvyihik5qb7";
+    })
+    # I should upstream this
+    ./post-install.patch
+  ];
+
+  postPatch = ''
+    chmod +x ./meson/post_install.py
```

Comment:
No need to prepend `./` to paths, unless you want to execute them. (And it would not be needed anyway, since the path already contains a slash.)